strawberryfields/tdm/tdmprogram.py

```diff
@@ -83,63 +84,136 @@ def validate_measurements(circuit, N):
     return spatial_modes
 
 
-def input_check(args, copies):
+def input_check(args):
     """Checks the input arguments have consistent dimensions.
 
     Args:
         args (Sequence[Sequence]): sequence of sequences specifying the value of the parameters
-        copies (int): number of times the circuit should be run
-
     """
     # check if all lists are of equal length
     param_lengths = [len(param) for param in args]
     if len(set(param_lengths)) != 1:
         raise ValueError("Gate-parameter lists must be of equal length.")
 
-    # check copies
-    if not isinstance(copies, int) or copies < 1:
-        raise TypeError("Number of copies must be a positive integer.")
-
 
-def _get_mode_order(num_of_values, N):
+def _get_mode_order(num_of_values, N, timebins):
     """Get the order by which the modes were measured"""
     all_modes = []
     for i in range(len(N)):
-        ra = list(range(sum(N[:i]), sum(N[: i + 1])))
-        all_modes.append(ra * ceil(max(N) / len(ra)))
+        modes = list(range(sum(N[:i]), sum(N[: i + 1])))
+        extended_modes = modes * ceil(1 + timebins // len(modes))
+        all_modes.append(extended_modes[:timebins])
 
     mode_order = [i for j in zip(*all_modes) for i in j]
-    mode_order *= ceil(num_of_values / len(mode_order))
+    mode_order *= ceil(1 + num_of_values / len(mode_order))
 
     return mode_order[:num_of_values]
 
 
-def reshape_samples(all_samples, modes, N):
+def reshape_samples(all_samples, modes, N, timebins):
     """Reshapes the samples dict so that they have the expected correct shape.
 
     Corrects the :attr:`~.Results.all_samples` dictionary so that the measured modes are the ones
     defined to be measured in the circuit, instead of being spread over a larger
     number of modes due to the mode-shifting occurring in :class:`~.TDMProgram`.
 
+    The function iterates through the unrolled circuit and populates, and returns, a new samples
+    dictionary with the shape ``{spatial mode: (shots, timebins)}``. E.g., this unrolled circuit:
+
+    .. code-block:: pycon
+
+        ...
+        MeasureHomodyne(0) | (q[0])  # shot 0, timebin 0, spatial mode 0  (sample 0)
+        MeasureHomodyne(0) | (q[2])  # shot 0, timebin 0, spatial mode 2  (sample 1)
+        ...
+        MeasureHomodyne(0) | (q[0])  # shot 0, timebin 1, spatial mode 0  (sample 2)
+        MeasureHomodyne(0) | (q[1])  # shot 0, timebin 1, spatial mode 2  (sample 3)
+        ...
+        MeasureHomodyne(0) | (q[0])  # shot 1, timebin 0, spatial mode 0  (sample 4)
+        MeasureHomodyne(0) | (q[2])  # shot 1, timebin 0, spatial mode 2  (sample 5)
+        ...
+        MeasureHomodyne(0) | (q[0])  # shot 1, timebin 1, spatial mode 0  (sample 6)
+        MeasureHomodyne(0) | (q[1])  # shot 1, timebin 1, spatial mode 2  (sample 7)
+
+    would return the dictionary
```

Comment:
How about including the equivalent `all_samples` that is produced by the above circuit?

Reply:
What do you mean? The dictionary in the example would be the `all_samples` that's returned by the above circuit (only with `(sample X)` instead of the actual sample value. Or do you mean the earlier `all_samples` that is passed to the `reshape_samples` function? :thinking:

Reply:
> do you mean the earlier all_samples that is passed to the reshape_samples function? thinking

Yes, this one.